cellxgene_resources/cellxgene_mods.py

```diff
@@ -636,3 +636,107 @@ def compare_revision(collection):
+    """
+    fm_dict = {'female': [female_ids, female_adata], 'male': [male_ids, male_adata]}
+    for k,v in fm_dict.items():
+        sex_specific_adata = v[1]
+        obs_data = pd.DataFrame(sex_specific_adata[:, sex_specific_adata.var.index].X.toarray(), columns=sex_specific_adata.var.index, index=adata.obs.index)
+        obs_donor_data = pd.merge(obs_data,adata.obs['donor_id'], how='left',left_index=True, right_index=True)
+        df = obs_donor_data.groupby(['donor_id'])[obs_donor_data.columns].sum(numeric_only=True).reset_index()
+        missing_genes = [g for g in v[0] if g not in df.columns]
+        df[missing_genes] = np.nan
+        v.append(df)
+    return fm_dict
+
+
+def assign_sex(x):
+    """
+    Input: ratio of male to female raw expression counts summed across all genes
+    Output: assignment of sex
+    """
+    if x > 0.35:
+        return 'male'
+    elif x < 0.05:
+        return 'female'
+    else:
+        return 'unknown'
+
+
+def check_percent(female_adata,male_adata,female_ids,male_ids):
+    """
+    Input: subset adatas and gene_id lists
+    Output: percent of genes found per sex
+    """
+    fem = female_adata.shape[1]/len(female_ids)
+    male = male_adata.shape[1]/len(male_ids)
+    print(f"% Female genes found: {(fem)*100}")
+    print(f"% Male genes found: {(male)*100}\n")
+    return (fem,male)
+
+
+def calculate_sex(fm_dict):
+    """
+    Input: output dictionary from generate_fm_dict()
+    Output: dataframe of both male and female donors that have 100+ total counts and stats
+    """
+    try:
+        female_df = fm_dict['female'][2]
+        female_df['female_sum'] = female_df.sum(numeric_only=True, axis=1)
+        male_df = fm_dict['male'][2]
+        male_df['male_sum'] = male_df.sum(numeric_only=True, axis=1)
+        male_female_df = pd.merge(male_df, female_df, how='left', on='donor_id')
+        male_female_df['total_sum'] = male_female_df[['female_sum','male_sum']].sum(numeric_only=True, axis=1)
+        donors_to_remove = male_female_df[male_female_df.total_sum < 100].donor_id.unique()  # Remove donors that have less than 100 total counts
+        if len(donors_to_remove) > 0:
+            print('Donors with < 100 total counts dropped: ', *(donors_to_remove), sep='\n')
+            male_female_df.drop(male_female_df[male_female_df.total_sum < 100].index, inplace=True)
+        #Calculate ratio and assign sex
+        male_female_df['male_to_female'] = male_female_df['male_sum']/male_female_df['female_sum']
+        male_female_df['scRNAseq_sex'] = male_female_df.apply(lambda x: assign_sex(x['male_to_female']), axis=1)
+        return male_female_df
+    except Exception as e:
+        print(e)
+
+
+def evaluate_donors_sex(adata):
+    genes_file = 'ref_files/sex_analysis_genes.json'
+    genes = json.load(open(genes_file))
+    female_ids = genes['female']
+    male_ids = genes['male']
+
+    fields_to_include = ['colllection_id','dataset_id','donor_id', 'sex_ontology_term_id']
```

Comment:
typo collection_id
but also neither that nor dataset_id are really needed since this is designed to run on any .h5ad

so you could probably remove this and just say metadata_list = ['donor_id','sex_ontology_term_id']